java/flight/flight-core/src/main/java/org/apache/arrow/flight/FlightService.java

```diff
@@ -88,12 +87,15 @@ public void listFlights(Flight.Criteria criteria, StreamObserver<Flight.FlightIn
   public void doGetCustom(Flight.Ticket ticket, StreamObserver<ArrowMessage> responseObserverSimple) {
     final ServerCallStreamObserver<ArrowMessage> responseObserver =
         (ServerCallStreamObserver<ArrowMessage>) responseObserverSimple;
+
     final GetListener listener = new GetListener(responseObserver, this::handleExceptionWithMiddleware);
-    try {
-      producer.getStream(makeContext(responseObserver), new Ticket(ticket), listener);
-    } catch (Exception ex) {
-      listener.error(ex);
-    }
+    executors.submit(() -> {
```

Comment:
I'd rather we not use the executor here unconditionally. A fully asynchronous implementation would set the callbacks and not need to execute in a separate thread at all; an implementation that wants to appear blocking but use the callbacks (as with BackpressureStrategy) should manage its own thread pool. We've had production issues when it turned out that the internal executor in FlightService was queueing requests without bound, defeating the rate limit/concurrency controls that had been placed on the gRPC executor.

Reply:
I think you'll have to change setOnReadyHandler below to call responseObserver.setOnReadyHandler itself

Reply:
Do you mean within GetListener? We register #onReady() on the responseObserver during GetListener's constructor, which will delegate to the onReadyHandler if it's been set.

Reply:
Sorry, yes, I meant within GetListener, but I suppose setting the handler unconditionally doesn't matter to gRPC - it'll never get called in the blocking case.